File: Crio/crio_programming_interview_problems-master/CheckSubtree/CheckSubtree.cpp
#include <bits/stdc++.h>
#include "../crio/cpp/ds/TreeNode/TreeNode.hpp"
#include "Solution.cpp"

using namespace std;
// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
int main()
{
    int tests;
    cin>>tests;
    while(tests--)
    {
        TreeNode* root  = TreeNode().readTreeReturnRoot();
        TreeNode* root2  = TreeNode().readTreeReturnRoot();
        bool answer = Solution().checkSubTree(root,root2);
        if(answer)
            cout<<"Yes\n";
        else
            cout<<"No\n";
    }
    return 0;
}

File: Crio/crio_programming_interview_problems-master/CheckSubtree/CheckSubtree.py
from crio.ds.Tree.TreeNode import readTreeReturnRoot
from Solution import *

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    testCases = int(input())
    for _t in range(testCases):
        root = readTreeReturnRoot()
        root2 = readTreeReturnRoot()
        if check_subtree(root, root2): 
           print("Yes")
        else: 
           print("No")

if __name__ == '__main__':
    import sys
    sys.setrecursionlimit(10 * 1000)
    main()

